File: frontend/src/types/product.ts
import { z } from 'zod';

export const ProductSchema = z.object({
  id: z.string(),
  name: z.string(),
  organizationId: z.string(),
  createdBy: z.object({
    id: z.string(),
    name: z.string()
  }),
  primaryImageUrl: z.string(),
  additionalImageUrls: z.array(z.string()),
  stage: z.enum(['queued', 'in_progress', 'completed', 'error']),
  log: z.string(),
  createdAt: z.string(),
  updatedAt: z.string(),
  backgroundRemovedImageUrl: z.string().nullable(),
  description: z.string().nullable(),
});

export type Product = z.infer<typeof ProductSchema>;

File: backend/api/response_types/product.py
from pydantic import BaseModel
from typing import List, Optional, Dict

class ProductResponse(BaseModel):
    id: str
    name: str
    organization_id: str
    created_by: Dict[str, str] = {
        "id": str,
        "name": str
    }
    primary_image_url: str
    additional_image_urls: List[str]
    stage: str
    log: str
    created_at: str
    updated_at: str
    background_removed_image_url: Optional[str] = None
    description: Optional[str] = None

class ProductsListResponse(BaseModel):
    products: List[ProductResponse]